lib/linalg.dx

```diff
@@ -144,35 +159,37 @@ def lu {n} (a: n=>n=>Float) :
   --      lRef!i!j := (a.i.j - s) / (get uRef!j!j)
   --    for i:n. ()
 
-    lmat = lower_tri_mat lRef
-    umat = upper_tri_mat uRef
+    -- Helper functions to index into triangular matrices
```

Comment:
I'm worried about these two, because their type does not guarantee their safety.  I would feel better if they were named unsafe_foo and documented what condition the user has to ensure to use them safely.

I think getting this right (or identifying the bugs in Dex that prevent us from getting it right) is pretty valuable, because it supports the story that Dex provides the indexing safety of `std::vector<float>` but the speed of `float*`; whereas if we find ourselves making increasingly long-range user-space arguments that this or that `unsafe_cast` is actually correct, that story degrades.

I assume you've tried for a while; if you're frustrated, do you want to set up a time to try to pair-program it?